protobuf/coreapi/api.proto

```diff
@@ -157,9 +157,23 @@ message ListenResultRequest {
 message ResultData {
   string executionID = 1;             // The unique identifier of the execution.
   string taskKey = 2;                 // The key of the executed task.
-  string outputKey = 3;               // The output's key from the returned task.
-  string outputData = 4;              // The output's data from the returned task, encoded in JSON.
   repeated string executionTags = 5;  // The list of tags associated with the execution
+  oneof result {
+    Output output = 6;   // output is filled when there is matching task execution that completed successfully.
+    Error error = 7;     // error is filled when there is a matching task execution that cannot be completed. 
+  }
+
+  message Output {
+    string key = 3;   // The output's key from the returned task.
+    string data = 4;  // The output's data from the returned task, encoded in JSON.
+  }
+  
+  message Error {
```

Comment:
`message Error` should be outside Result Data as it will be used for diffrent error (like for ListenEvent)

Reply:
I agree. We can create custom message in gRPC and reuse them anywhere

Reply:
Let's do this when we need it. It won't be a breaking change when we do this in future. Because we may not need to use _Error_ inside _ListenEvent_. We should first check to see if we need to have errors inside listen event like we need it here.

Reply:
@NicolasMahe We may extend this specific _Error_ type with another fields like _type_, _data_ that talked about other comments in this PR. So I think that we may not be able to use a generic error type here.

Reply:
Can I get more feedbacks about the direction of this discussion?

Reply:
Ok, as you said, let's do it when we need it if it will not break anything to move it later ;)